narwhal/primary/src/tests/core_tests.rs

```diff
@@ -2,445 +2,12 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 use super::*;
-use crate::common::{create_db_stores, create_test_vote_store};
-use anemo::{types::PeerInfo, PeerId};
+use crate::common::create_db_stores;
+
 use fastcrypto::traits::KeyPair;
 use prometheus::Registry;
-use test_utils::{fixture_batch_with_transactions, CommitteeFixture, PrimaryToPrimaryMockServer};
+use test_utils::CommitteeFixture;
 use tokio::time::Duration;
-use types::{CertificateDigest, Header, LatestHeaderResponse, MockPrimaryToPrimary, Vote};
-
```

Comment:
Apologies if that already exists and I am missing this, but could you please add a test that covers the case of the processing of a header that our node proposes?

Reply:
Added